### PR TITLE
removes Box<dyn Iterator<...>> from rpc/src/rpc_subscriptions.rs

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
`Box<dyn ...>` requires dynamic dispatch, is heap allocated, slow and verbose.

#### Summary of Changes
removed `Box<dyn Iterator<...>>`

### CODE REVIEW

File: rpc/src/rpc_subscriptions.rs [hunk 1]
```diff
@@ -430,40 +418,36 @@ fn filter_program_results(
...
+    } else {
+        let accounts = keyed_accounts.map(move |(pubkey, account)| RpcKeyedAccount {
+            pubkey: pubkey.to_string(),
+            account: UiAccount::encode(&pubkey, &account, encoding, None, None),
+        });
+        (Some(accounts), None)
+    };
+    let accounts = accounts.into_iter().flatten();
+    let token_accounts = token_accounts.into_iter().flatten();
+    (accounts.chain(token_accounts), last_notified_slot)
```
Comment (CriesofCarrots) [inline rpc/src/rpc_subscriptions.rs:436]: This one is pretty hard to follow now. I guess the issue is with iterator types, but is there no better way to write this?

**Review by CriesofCarrots — COMMENTED** [review]
Generally I approve; just wonder if we can make the `filter_program_results()` impl less bad...?

File: rpc/src/rpc_subscriptions.rs (hunk 1 above)
Comment (behzadnouri) [reply rpc/src/rpc_subscriptions.rs]: Fair point. Switched to `Itertools::Either`. wdyt?

Comment (CriesofCarrots) [reply rpc/src/rpc_subscriptions.rs]: Way better, thanks!

**Review by CriesofCarrots — APPROVED** [review]
